Destruction: The Game

danjwuest opened the pull request:
All tests pass.

danjwuest committed:
Initial Table Creation

danjwuest committed:
players table syntax fix

danjwuest committed:
foreign key constraints cleanup

danjwuest committed:
final foreign key constraint fix

danjwuest committed:
Character Items first

danjwuest committed:
Character Items View syntax

danjwuest committed:
Double Corrected character items view syntax

danjwuest committed:
Character Items with union

danjwuest committed:
Ordering character items by name

danjwuest committed:
team items view + rename of character_id column in character items

danjwuest committed:
team_items joined to team_members, then to teams

danjwuest committed:
team_items join team_members-> tm

danjwuest committed:
team_items rework without referencing character items view

danjwuest committed:
removed GROUP BY in team_items view

danjwuest committed:
Adding armor_total

danjwuest committed:
Adding equip and unequip

danjwuest committed:
include character stats in armor_total

danjwuest committed:
Add attack procedure, health no longer unsigned

danjwuest committed:
Updated attack to include items table alias "i"

danjwuest committed:
Attack procedure: Changed syntax to correctly output total armor data

danjwuest committed:
Re-paste of attack procedure

It works in the workbench, so I'm not sure why it isn't here. Repasting to double check any possible typos.

danjwuest committed:
Updated attack to reference equip id instead of item id

danjwuest committed:
Update attack to correctly compare equipped_id

danjwuest committed:
Adding set_winners

halverneus commented on week7/answers.sql:
```diff
@@ -1,3 +1,284 @@
+
+CREATE TABLE players (
+  player_id INT UNSIGNED PRIMARY KEY AUTO_INCREMENT,
+  first_name VARCHAR(30),
```
Comment througout: be sure to include NOT NULL (or you might find some math breaking if someone were to insert a NULL value).

danjwuest replied:
NOT NULL has been added,

halverneus commented on week7/answers.sql:
Don't use "SELECT *". If the table is ever altered, this code will break. This will also save you from needing a variable for "equipping_inventory_id".

(Same comments for "unequip()".)

danjwuest replied:
"SELECT *" has been changed to specific fields.

halverneus commented on week7/answers.sql:
Only an optimization and not required: if you use the UPDATE in the ELSE clause of the IF statement you use for DELETE FROM, then you can save a call to UPDATE in cases where a character simply dies.

danjwuest replied:
Interesting, that gracefully handles character death without an extra trip to the database. I have now included it.

halverneus commented on week7/answers.sql:
This is correctly done, no CURSOR required.

halverneus reviewed (approved):
You are welcome to update your code based on the feedback and the automated test results and resubmit the assignment. If you are satisfied with your grade, go ahead and merge this PR.

danjwuest committed:
Correction from feedback

General: Most/all fields are now not allowed to be null.
Equip and Unequip: Fields are now specified when pulling data into variables.
Attack: Character health is no longer updated if the health will be 0 or less. Instead, the character just dies.

danjwuest commented:
I have implemented all the changes from the feedback, please let me know if there's anything else. Resubmitting now.

halverneus reviewed (approved):
Looks great! Feel free to merge your Pull Request.
